### PR TITLE
Add sample for dumping object functions

### LINKED ISSUE #149: classのインスタンスをそのまま保存して後でロードする方法
などありますか。特にrobot-interfaceクラスのインスタンスに対して使いたいです。

やりたいことは、ロボットのステート（認識結果やアングルベクターなど）がslotsに入っているので、
testを行うときにそのままロードして各メソッドのテストをする、ということです。
rosbagを再生してもいいですが、さっとできる方がいいのと、
あるメソッドのための初期姿勢に移動するために準備（必要なメソッドを順に呼ぶ）が必要なのをどうにかしたいというものです。

### PR DESCRIPTION
**Pull request by wkentaro**
Closes #149

いまいち`dump-object`と`dump-structure`の違いがわからないので適切か見てください。

### CODE REVIEW

**Comment by k-okada** [comment]
https://github.com/euslisp/EusLisp/blob/master/lisp/l/loader.l#L360
```
(defun dump-object (file &rest objects)
   (let
      ((*print-object* t) (*print-circle* t)
	(*print-level* nil) (*print-length* nil))
      (if (streamp file)
	  (dolist (obj objects) (print obj file))
	  (with-open-file (f file :direction :output)
             (dolist (obj objects) (print obj f))))))

(defun dump-structure (file &rest objects)
   (let
      ((*print-structure* t) (*print-circle* t)
	(*print-level* nil) (*print-length* nil))
      (if (streamp file)
	  (dolist (obj objects) (print obj file))
	  (with-open-file (f file :direction :output)
             (dolist (obj objects) (print obj f))))))
```
同じ？

**Comment by k-okada** [comment]
```
<DT><STRONG>*print-circle*</STRONG></DT>
<DD>再帰的参照を残したオブジェクトを印刷する。
</DD>
<DT><STRONG>*print-object*</STRONG></DT>
<DD>全てのオブジェクトの詳細を印刷する。
</DD>
<DT><STRONG>*print-structure*</STRONG></DT>
<DD><TT>#</TT>書式を使ってオブジェクトを印刷する。
</DD>
```
in 
https://github.com/euslisp/EusLisp/blob/bbae82c470932d17260d8a34f1646ae6c05f4143/doc/html/jmanual-node56.html

**Comment by k-okada** [comment]
で，
https://github.com/euslisp/EusLisp/blob/74fff34afa261db392800f1f2ea7518ee4cadfff/lisp/c/printer.c#L584
```
    else if (Spevalof(PROBJECT)!=NIL)    prinxobj(ctx,x,f,fobj,prlevel-1);
    else if (pisarray(x) && (classof(x)==C_ARRAY))  printarray(ctx,x,f,prlevel-1);
    else if (Spevalof(PRSTRUCTURE)!=NIL) printstructure(ctx,x,f,fobj,prlevel-1);
```

から，
```
static void printstructure(ctx,x,f,fobj,prlevel)
register context *ctx;
register pointer x,f;
register pointer fobj;
int prlevel;
{ pointer klass, *varvec;
  register int i=0,s;
  int prlength=getprlength(ctx);
  
  writestr(f,(byte *)"#s(",3);
  klass=classof(x);
  printsym(ctx,klass->c.cls.name,f);
  s=objsize(x); varvec=klass->c.cls.vars->c.vec.v;
  prlength--;
  while (i<s && prlength>0) {
    writech(f,' ');
    printsym(ctx,varvec[i],f);
    writech(f,' ');
    if (i==0) prin1(ctx,fobj,f,prlevel);
    else prin1(ctx,x->c.obj.iv[i],f,prlevel);
    i++; prlength-=2;}
  if (i<s) writestr(f,(byte *)" ...",4);
  writech(f,')');
  }

static void prinxobj(ctx,x,f,fobj,prlevel)
register context *ctx;
register pointer x,f;
pointer fobj;
int prlevel;
{ pointer klass;
  register int i=0,s;
  int prlength=getprlength(ctx);
  writestr(f,(byte *)"#J(",3);
  klass=classof(x);
  printsym(ctx,klass->c.cls.name,f);
  s=objsize(x);
  while (i<s && prlength-->0) {
    writech(f,' ');
    if (i==0) prin1(ctx,fobj,f,prlevel);
    else prin1(ctx,x->c.obj.iv[i],f,prlevel);
    i++;}
  writech(f,')');
  }
```
https://github.com/euslisp/EusLisp/blob/74fff34afa261db392800f1f2ea7518ee4cadfff/lisp/c/printer.c#L501

**Comment by k-okada** [comment]
#s はcommonlisp由来 #j は euslisp?

**Comment by wkentaro** [comment]
結局同じという結論でしょうか．．

**Comment by wkentaro** [comment]
Please merge.

File: doc/latex/evaluation.tex [hunk 1]
```diff
@@ -694,6 +694,29 @@ \subsection{Dump Objects}
...
+(dump-object "a_cube.l" a)
+(with-open-file
+  (f "a_cube.l" :direction :input)
+  (setq a (read f)))
+(print a)
+
+;; sample for dump-structure
+(dump-structure "a_cube.l" a)
+(with-open-file
+  (f "a_cube.l" :direction :input)
```
Comment (k-okada) [inline doc/latex/evaluation.tex]: `a-cube.l` seems better

Comment (wkentaro) [inline doc/latex/evaluation.tex]: Hoops, I will change that.
Anyway, are there any file name extension for dumped object?
like .pkl for pickle in Python
and .mat in Matlab?

Comment (wkentaro) [inline doc/latex/evaluation.tex]: why not .ldump?

**Comment by wkentaro** [comment]
ping

**Pull request merged**